📊 feat: Improve Helm Chart

hofq opened the pull request:
# Pull Request Template

⚠️ Before Submitting a PR, Please Review:
- Please ensure that you have thoroughly read and understood the [Contributing Docs](https://github.com/danny-avila/LibreChat/blob/main/.github/CONTRIBUTING.md) before submitting your Pull Request.

⚠️ Documentation Updates Notice:
- Kindly note that documentation updates are managed in this repository: [librechat.ai](https://github.com/LibreChat-AI/librechat.ai)

# Summary
For now, it's a draft to document my progress.
CC: @danny-avila, @iamNoah1 

## Done
- Subcharts for Dependencies
- Create RAG_API Subchart
- Provide full getting started Guide
- Keep Chart 1:1 Compatible to old Chart/ Document Changes with Version Change
- Move Guide to Docs
## Todo

- Add Rag_Api Guide


# Change Type

Please delete any irrelevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update
- [ ] Translation update

# Testing

Please describe your test process and include instructions so that we can reproduce your test. If there are any important variables for your testing configuration, list them here.

## **Test Configuration**:

# Checklist

Please delete any irrelevant options.

- [x] I have performed a self-review of my own code
- [X] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [ ] I have written tests demonstrating that my changes are effective or that my feature works
- [ ] Any changes dependent on mine have been merged and published in downstream modules.
- [x] A pull request for updating the documentation has been submitted.

hofq committed:
Replaced Helm Charts with Blue Atlas Charts

hofq committed:
Fix Workflow

hofq committed:
improve docs

hofq committed:
update gitignore

hofq committed:
Update docs

hofq commented:
@danny-avila @iamNoah1 
I'm struggling a bid of choosing a way of the config syntax.

So in our chart for example we chose to place env config under librechat.configEnv and in the official chart its config.env.

No big deal, but everyone that uses the other chart has to manually migrate his config to the new format. Should I decide based on official/unofficial, user count of the chart, or make some dual-solution that just takes both config variants (maybe with an additional deprecation warning)?

hofq committed:
change values order, add hpa

hofq committed:
change tls example domain

hofq committed:
Default: Enable liveness and readiness

danny-avila commented:
> Should I decide based on official/unofficial, user count of the chart, or make some dual-solution that just takes both config variants (maybe with an additional deprecation warning)?

I think this is pretty recent so whatever you choose shouldn’t be too disruptive. Maybe both just to be safe but I don’t think it’s totally necessary

iamNoah1 commented:
@hofq I don't have a strong opinion here.

iamNoah1 commented on helm/librechat/values.yaml:
```diff
@@ -0,0 +1,215 @@
+  #         modelDisplayLabel: "OpenRouter"
+
+
+# # name of existing Yaml configmap, key must be librechat.yaml
```
```suggestion
# name of existing Yaml configmap, key must be librechat.yaml
```

iamNoah1 commented on helm/librechat/values.yaml:
```diff
@@ -0,0 +1,215 @@
+fullnameOverride: ""
+
+librechat:
+# LibreChat allows Configuration in 2 Ways: Environment Variables and a Config file. For easier Deployment the needed values are predifined here but should be adjusted to your needs.
```
This should go into docs if not already present

hofq replied:
as above it should be documented both in docs and here

iamNoah1 replied:
agreed

iamNoah1 commented on helm/librechat/values.yaml:
```diff
@@ -0,0 +1,215 @@
+
+
+  
+  # # For adding a custom config yaml-file you can set the contents in this var. See https://www.librechat.ai/docs/configuration/librechat_yaml/example
```
```suggestion
  # For adding a custom config yaml-file you can set the contents in this var. See https://www.librechat.ai/docs/configuration/librechat_yaml/example
```

iamNoah1 commented on helm/librechat/templates/configmap-env.yaml:
```diff
@@ -0,0 +1,17 @@
+  {{- if (index .Values "librechat-rag-api" "enabled") }}
+  RAG_API_URL: http://{{ include "rag.fullname" (index .Subcharts "librechat-rag-api") | lower }}.{{ .Release.Namespace | lower }}.svc.cluster.local:8000 
+  {{- end }}
+  {{- if .Values.librechat.configEnv }}
```
It is just cosmetic, but I would put them more likely at the end than in between

iamNoah1 commented on helm/librechat/values.yaml:
```diff
@@ -0,0 +1,215 @@
+nameOverride: ""
+fullnameOverride: ""
+
+librechat:
```
Generally I am a bit confused why we have global.librechat and librechat?! Feels a bit redundant, can we merge them and have only one? Preferably only librechat on the root level.

hofq replied:
global is a directive defined by helm. If there are values that should be available everywhere you opt for the global directive. The Secret Names and API Key may also be used in the subchart

iamNoah1 replied:
Could we the librechat stuff from the root level then also to the global.librechat?

hofq replied:
hmm don't know if i'd like that. would be a bid of for moving every var in global just for 2 secret names that have to be declared here. maybe we could use yaml anchors and move the vars into the normal librechat block?

iamNoah1 reviewed (commented):
Wow, I made it through ^^ It took forever, sorry. But I think we can and should move on now :) Happy to help any further. @hofq what you think?!

hofq commented:
Thanks @iamNoah1 :) I will look into it if i find time to rebase that whole pr. There where some Changes to our chart repo in the past which also should be included here. 


Additionally maybe putting renovate on the table for that PR would also be nice

hofq reviewed (commented):
Just some notes, there are some changes already done in https://github.com/bat-bs/helm-charts which i have to adapt.

Thanks for the Review :)

iamNoah1 commented on helm/librechat-rag-api/Chart.yaml:
```diff
@@ -0,0 +1,33 @@
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.5
```
understood, makes sense

iamNoah1 commented on helm/librechat-rag-api/readme.md:
```diff
@@ -0,0 +1,4 @@
+# Librechat RAG API Helm CHart
```
understood

iamNoah1 commented on helm/librechat/values.yaml:
```diff
@@ -0,0 +1,215 @@
+  librechat:
+    # existing Secret for all envs/ only Passwords. Can be locally generated with: kubectl create secret generic librechat-secret-envs --from-env-file=.env.example --dry-run=client -o yaml > secret-envs.yaml
+    ## For better maintainabillity, you can put all vars directly in the config Section and only overwrite Secrets with this if nessesary.
+    # Required Values:
```
agreed

Commit:
chore: bump base chart

Commit:
apply requested changes

hofq commented on helm/librechat-rag-api/templates/rag-deployment.yaml:
```diff
@@ -0,0 +1,85 @@
+ 
+
+
+      volumes:
```
added templating for that, as some user requested it. see https://github.com/bat-bs/helm-charts/pull/49

hofq committed:
Merge branch 'main' into blue_atlas_helm

Commit:
add Release fix

Commit:
add: error handling

hofq commented:
https://github.com/LibreChat-AI/librechat.ai/pull/231

hofq commented:
Hi Sorry for my Late Comment and Changes :/

I'm no longer employed with Blue Atlas as the Company will not persist to exist. However i will assist and move the Migration of our Chart forward in my free time.

I added most of your suggested changes @iamNoah1 bumped our base chart and added error handling. Hope you will find time to have a look again :)

iamNoah1 commented on helm/librechat-rag-api/values.yaml:
```diff
@@ -0,0 +1,93 @@
+  enabled: true
+  # nameOverride: vectordb
+  image:
+    registry: ghcr.io
```
never mind ^^

iamNoah1 reviewed (commented):
@hofq at this point I would say lets get it merged and we can still adjust afterwards. Only thing I would ask you to check is if the docs are up to date so that people could use it :) Ah and there are merge conflicts...

danny-avila commented:
Is this ready? Will check back in a week.

hofq commented:
I will test it and bump the Version, after that we should be set :)
Thanks

iamNoah1 commented:
friendly ping @hofq

hofq committed:
Merge branch 'main' into blue_atlas_helm

Commit:
chore: cleanup and testing

Commit:
fix: adjust Chart.yaml

hofq commented:
All Done here :D

https://github.com/LibreChat-AI/librechat.ai/pull/231

Please, check if my Documentation is readable enough to deploy and let me know what's missing

rubentalstra committed:
Merge branch 'main' into blue_atlas_helm

iamNoah1 commented:
@danny-avila we good to go :)

hofq committed:
Merge branch 'main' into blue_atlas_helm

danny-avila commented:
Thanks for all the hard work on this!

ion-elgreco commented:
@AurimasNav I just copied the charts in main to my repo, I guess they didnt setup a release pipeline for it yet 🤷

ion-elgreco commented:
@danny-avila then it would be wise to remove this wording from the docs, as it seems even the migrated one won't be maintained.

> [LibreChat Helm Chart by Blue Atlas Helm Charts](https://github.com/bat-bs/helm-charts/tree/main/charts/librechat) # will be depricated soon as its migrated here

ion-elgreco commented:
I am already using the helm chart, but just copied from this repo directly. If you need any help maintaining this, I can do so, but I need some clarity first on why they are not released? Is it because you don't want to maintain it or there is simply no CI pipeline?

danny-avila commented:
@ion-elgreco read the link I shared, it should answer your question

ion-elgreco commented:
> @ion-elgreco read the link I shared, it should answer your question

Yeah that thread is not clear tbh, that's why I'm asking

iamNoah1 commented:
I think it is a bit of both. With the current setup they should be released in this github repo but they are not and no one seems to know exactly why. For the Docker images it works, for the helm charts not. So guess everyone would be happy if somehow could help to make the pipeline work.

hofq commented:
@ion-elgreco @iamNoah1 @danny-avila 

There is nothing to be fixed here.

Github Pages is not working correctly.

Danny, can you try to delete the Branch gh-pages and rerun? 

i do not have any permissions on this Repository

hofq commented:
if there comes any request up regarding features of the chart i would be happy to help. Its not my Repository so i just can't help on this one.

I wrote this in another thread too, its the same pipeline as on bat bs

hofq commented:
Hi @ion-elgreco 

is there anything missing in the current form, cause my old one should still work fine and is on feature parity.
https://github.com/bat-bs/helm-charts

I try to work with Danny on it as soon as i hear back from him

ion-elgreco commented:
> Hi @ion-elgreco
> 
> is there anything missing in the current form, cause my old one should still work fine and is on feature parity. https://github.com/bat-bs/helm-charts
> 
> I try to work with Danny on it as soon as i hear back from him

Adding extraEnvs from secretReferences
